File: PHP/Program/Kurir.php
<?php
// Kurir.php
require_once __DIR__ . '/Pengiriman.php';

class Kurir extends Pengiriman {
    private $namaKurir;
    private $statusPengiriman;
    private $noHP;
    private $photoFilename; // nama file foto (opsional)

    public function setNamaKurir(string $nama) {
        $this->namaKurir = $nama;
    }

    public function getNamaKurir(): string {
        return $this->namaKurir ?? '';
    }

    public function setStatusPengiriman(string $status) {
        $this->statusPengiriman = $status;
    }

    public function getStatusPengiriman(): string {
        return $this->statusPengiriman ?? '';
    }

    public function setNoHP(string $noHP) {
        $this->noHP = $noHP;
    }

    public function getNoHP(): string {
        return $this->noHP ?? '';
    }

    public function setPhotoFilename(string $name) {
        $this->photoFilename = $name;
    }

    public function getPhotoFilename(): string {
        return $this->photoFilename ?? '';
    }

    // Convert object to array for saving
    public function toArray(): array {
        return [
            'nama' => $this->getNama(),
            'email' => $this->getEmail(),
            'metode' => $this->getMetode(),
            'barang' => $this->getBarangList(),
            'alamat' => $this->getAlamat(),
            'ekspedisi' => $this->getEkspedisi(),
            'waktu' => $this->getWaktuPengiriman(),
            'namaKurir' => $this->getNamaKurir(),
            'status' => $this->getStatusPengiriman(),
            'noHP' => $this->getNoHP(),
            'photo' => $this->getPhotoFilename()
        ];
    }

    // Create object from array (for loading)
    public static function fromArray(array $a): Kurir {
        $k = new Kurir();
        $k->setPembelian($a['nama'] ?? '', $a['email'] ?? '');
        $k->setMetode($a['metode'] ?? '');
        if (!empty($a['barang']) && is_array($a['barang'])) $k->setBarangArray($a['barang']);
        $k->setAlamat($a['alamat'] ?? '');
        $k->setEkspedisi($a['ekspedisi'] ?? '');
        $k->setWaktuPengiriman($a['waktu'] ?? '');
        $k->setNamaKurir($a['namaKurir'] ?? '');
        $k->setStatusPengiriman($a['status'] ?? '');
        $k->setNoHP($a['noHP'] ?? '');
        $k->setPhotoFilename($a['photo'] ?? '');
        return $k;
    }
}

File: Python/Program/Kurir.py
from Pengiriman import Pengiriman

# Kelas Kurir turunan dari Pengiriman
class Kurir(Pengiriman):
    def __init__(self):
        super().__init__()
        self.nama_kurir = ""           # Nama kurir
        self.status_pengiriman = ""    # Status pengiriman
        self.no_hp = ""                # Nomor HP kurir

    def set_nama_kurir(self, nama):
        self.nama_kurir = nama         # Setter nama kurir

    def get_nama_kurir(self):
        return self.nama_kurir         # Getter nama kurir

    def set_status_pengiriman(self, status):
        self.status_pengiriman = status    # Setter status pengiriman

    def get_status_pengiriman(self):
        return self.status_pengiriman      # Getter status pengiriman

    def set_no_hp(self, no_hp):
        self.no_hp = no_hp             # Setter nomor HP kurir

    def get_no_hp(self):
        return self.no_hp              # Getter nomor HP kurir
